src/pages/index.js

```diff
@@ -5,77 +5,210 @@ import PopupWithImage from "../components/PopupWithImage.js";
+const profileImageButton = document.querySelector(".profile__icon");
+const avatarModal = document.querySelector("#profile-picture-modal");
+const avatarModalButton = avatarModal.querySelector(".modal__button");
+// const deleteButton = document.querySelector(".elements__trash-button");
 
 //Section Render
 
-const cardsSection = new Section(
+// const cardsSection = new Section(
+//   {
+//     renderer: renderCard,
+//   },
+//   ".elements__cards"
+// );
+
+//User Info Render
+
+const userInfo = new UserInfo(
+  ".profile__name",
+  ".profile__description",
+  ".profile__image"
+);
+
+//API Render
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7b855de8-b589-492b-9e1c-b631d0b0f4c2",
+    "Content-Type": "application/json",
+  },
+});
+
+const cardList = new Section(
   {
-    items: initialCards,
-    renderer: renderCard,
+    renderer: (cardData) => {
+      cardList.addItem(createCard(cardData));
+    },
   },
   ".elements__cards"
 );
 
+api
+  .getInitialCards()
+  .then((res) => {
+    // console.log(">>Initial Cards", cardData);
+    cardList.renderItems(res.reverse());
+    console.log(res);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    userInfo.setUserAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 //FUNCTIONS----------------------
 
+function setButtonText(button, text) {
+  button.textContent = text;
+}
+
 function createCard(cardData) {
-  const card = new Card(cardData, "#element-template", () =>
-    handleImageClick(cardData)
+  const card = new Card(
+    cardData,
+    "#element-template",
+    () => handleImageClick(cardData),
+    handleDeleteButton,
+    handleCardLike
   );
   return card.getView();
 }
 
-function renderCard(cardData) {
-  const card = createCard(cardData);
-  cardsSection.addItem(card);
-}
+// function renderCard(cardData) {
+//   const card = createCard(cardData);
+//   section.addItem(card);
+// }
+
+//Image Popup Render
+
+const imagePopup = new PopupWithImage("#modal-preview");
+imagePopup.setEventListeners();
 
 function handleImageClick(cardData) {
   imagePopup.open(cardData.name, cardData.link);
 }
 
-cardsSection.renderItems();
+function handleCardLike(card) {
+  api
+    .likeCard(card._id, card.isLiked())
+    .then(() => {
+      card.handleLikeIcon();
+    })
+    .catch(console.error);
+}
 
-//User Info Render
+function handleProfileEditSubmit({ name, about }) {
+  editProfilePopup.setLoading(true);
+  return api
+    .updateProfileInfo(name, about)
+    .then((userData) => {
+      userInfo.setUserInfo(userData);
+      editProfilePopup.close();
+    })
+    .finally(() => {});
+}
 
-const userInfo = new UserInfo(".profile__name", ".profile__description");
+function handleAddNewCardSubmit(cardData) {
+  // renderCard(cardData, cardsWrap);
```

Comment:
Needs correction: you should also notify the user about loading process by changing the button text